sources/PseudoTerminal.m

```diff
@@ -5637,6 +5637,16 @@ - (IBAction)moveTabRight:(id)sender
     [self tabsDidReorder];
 }
 
+- (IBAction)expandRow:(id)sender {
+    [self.currentSession.screen terminalSetRows:self.currentSession.rows + 1
```

Comment:
If you call `[self sessionInitiatedResize:self.currentSession width:self.currentSession.columns height:self.currentSession.rows+1]` then it'll work even if terminal-initiated resizing is disabled. Same in `shrinkRow`.